### DISCUSSION TITLE
CircleLayer

### DISCUSSION THREAD

**Discussion by Kovalskiii** [rnmapbox/maps#2931] [Q&A]
I use CircleLayer with offeset for positioning under point count. But when i zoom out, Circle is moving left too, it is not fixed. How do i to can fix this behavior ?
 
`  <Mapbox.CircleLayer
id="clusteredMarkerIconCircle"
belowLayerID="clusteredMarkersIcon"
filter={['has', 'point_count']}
style={styles.clusteredMarkersIconCircle}
 />`

`             <Mapbox.SymbolLayer
                id="clusteredMarkerCount"
                style={styles.markerCount}
                filter={['has', 'point_count']}
              /> 
`

`markerCount: {
    textField: '{point_count}',
    textSize: 12,
    textPitchAlignment: 'map',
    textOffset: [2, -1.8],
    textColor: 'white',
    textAllowOverlap: true,
  },
`
`
  markerCountCircle: {
    circleColor: '#8c4fef',
    circleRadius: 10,
    circleStrokeWidth: 1,
    circleStrokeColor: 'black',
    circleTranslateAnchor: 'map',
    circleTranslate: [24, -22]
  },
`

**Comment by Kovalskiii** [comment]
![image](https://github.com/rnmapbox/maps/assets/49062638/05117d41-1d4d-4910-9bf6-2dfd80d3c860)
![image](https://github.com/rnmapbox/maps/assets/49062638/df876e4e-0d33-495f-bd79-2aee9e248d7d)


@mfazekas Is it bug, or i something missed ?

**Comment by mfazekas** [comment]
@Kovalskiii sorry I'm not sure, have you tried viewport as anchor? Have you tested in in Mapbox gl js?

**Reply by Kovalskiii** [reply]
I tried 'viewport', same. Note, i use globe mode, in mercator i do not have that problem

**Reply by Kovalskiii** [reply]
I need solution

**Comment by Kovalskiii** [comment]
@mfazekas  Something wrong with circleTranslate style parameter, because if i do not use it, circle ramains on the coordinate, no mater i zoom in or out, but if i use circleTranslate with some offset, happens what i discribed previously.

**Comment by Kovalskiii** [comment]
@mfazekas  Do i can to get points count from ShapeSource ref for every cluster ? In order to use View for showing points count text without SymbolLayer.
